e2e/e2e_test.go

```diff
@@ -63,7 +80,8 @@ func simpleReadWrite(errChan chan error, outChan chan string, conn io.ReadWriteC
 */
 func TestPionE2ESimple(t *testing.T) {
 	expectedGoRoutineCount := runtime.NumGoroutine()
-	serverPort := baseServerPort
+
+	serverPort := pickPort(t)
```

Comment:
I would move this to line 77, instead of `serverPort++` do `serverPort == pickPort(t)`

Great fix though, thanks for grabbing :)